src/main/java/backend/UpdateController.java

```diff
@@ -146,4 +152,46 @@ private void openRepositoriesInFilters(List<FilterExpression> filterExprs) {
         // No sort or updated, return sort by descending ID, which is the default.
         return Qualifier.getSortComparator(models, "id", true, false);
     }
+
+    /**
+     * Constructs GUIElements (including all necessary references to labels/milestones/users to properly display
+     * the issue) corresponding to a list of issues without changing the order.
+     *
+     * @param models The MultiModel from which necessary references are extracted.
+     * @param filteredAndSortedIssues The list of issues to construct GUIElements for.
+     * @return A list of GUIElements corresponding to the given list of issues.
+     */
+    private List<GuiElement> produceGUIElements(MultiModel models, List<TurboIssue> filteredAndSortedIssues) {
+        return filteredAndSortedIssues.stream().map(issue -> {
+            Optional<Model> modelOfIssue = models.getModelById(issue.getRepoId());
+            assert modelOfIssue.isPresent();
+
+            return new GuiElement(issue,
+                    produceRelevantLabels(modelOfIssue.get(), issue),
+                    models.getMilestoneOfIssue(issue),
+                    models.getAssigneeOfIssue(issue));
+        }).collect(Collectors.toList());
+    }
+
+    /**
+     * Creates a list of relevant labels to an issue, containing current labels of the issue combined with
+     * labels removed from the issue.
+     *
+     * @param model The model containing the relevant issue.
+     * @param issue The issue, the labels of which are to be retrieved.
+     * @return A list of labels needed for displaying the issue properly.
+     */
+    private List<TurboLabel> produceRelevantLabels(Model model, TurboIssue issue) {
+        Stream<TurboLabel> removedLabels = issue.getMetadata().getEvents().stream()
+                .filter(issueEvent -> issueEvent.getType() == IssueEventType.Unlabeled)
+                .map(unlabeledEvent -> {
+                    Optional<TurboLabel> label = model.getLabelByActualName(unlabeledEvent.getLabelName());
+                    assert label.isPresent() : issue + " Label \"" + unlabeledEvent.getLabelName() + "\" not found.";
```

Comment:
Hmm, just to confirm -- it's not possible that we trigger this assertion because of an old label being deleted or renamed on GitHub, right? If metadata is showing, it must be up to date, so we would never have an old label in the events?